TUnit.Assertions/AssertConditions/BecauseReason.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Globalization;
+
+namespace TUnit.Assertions.AssertConditions;
+
+internal class BecauseReason(string because, object[] becauseArgs)
+{
+    private string? _message;
+
+    private string CreateMessage()
+    {
+        const string prefix = "because";
+        try
+        {
+            string message = because.Trim();
+
+            if (becauseArgs?.Length > 0)
+            {
+                message = string.Format(CultureInfo.InvariantCulture, message, becauseArgs);
+            }
+
+            return !message.StartsWith(prefix, StringComparison.OrdinalIgnoreCase)
+                ? $" {prefix} {message}"
+                : $" {message}";
+        }
+        catch (FormatException)
+        {
+            return $"**WARNING: the because message '{because}' could not be formatted correctly!**";
+        }
+    }
+
+    public override string ToString()
+    {
+        _message ??= CreateMessage();
+        return _message;
+    }
+}
```

Comment:
Do you want to create implicit operators to automatically convert to and from strings?

Reply:
I didn't intend to. This class is only an internal class to avoid sanitizing the reason multiple times for multiple assertions.

Reply:
Ah right! That's fine then